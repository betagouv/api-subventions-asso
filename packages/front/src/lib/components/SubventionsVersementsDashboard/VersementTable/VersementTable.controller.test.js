import VersementTableController from "./VersementTable.controller";
import VersementsAdapter from "$lib/resources/versements/versements.adapter";

vi.mock("$lib/resources/versements/versements.adapter", () => ({
    default: {
        toVersement: vi.fn(() => ({
            totalAmount: undefined,
            centreFinancier: undefined,
            lastVersementDate: undefined,
        })),
    },
}));

describe("VersementTableController", () => {
    describe("extractHeaders()", () => {
        it("return an array of header", () => {
            const actual = VersementTableController.extractHeaders();
            expect(actual).toMatchSnapshot();
        });
    });

    describe("extractRows()", () => {
        it("should call _extractTableDataFromElement for each element in array", () => {
            VersementTableController.extractRows([{ versements: [{}] }, { versements: [{}] }]);
            expect(VersementsAdapter.toVersement).toHaveBeenCalledTimes(2);
        });

        it("should not call _extractTableDataFromElement if element has no versements", () => {
            VersementTableController.extractRows([{}, { versements: [{}] }]);
            expect(VersementsAdapter.toVersement).toHaveBeenCalledTimes(1);
        });

        it("should return an array", () => {
            const expected = [null, null];
            const actual = VersementTableController.extractRows([{}, {}]);
            expect(actual).toEqual(expected);
        });
    });
});
