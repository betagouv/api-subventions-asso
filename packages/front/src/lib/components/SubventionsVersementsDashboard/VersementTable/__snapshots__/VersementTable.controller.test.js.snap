// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VersementTableController > extractHeaders() > return an array of header 1`] = `
[
  "Montant versÃ©",
  "Centre financier",
  "Date du versement",
  "BOP",
]
`;

exports[`VersementTableController extractHeaders() return an array of header 1`] = `
[
  "Montant versÃ©",
  "Centre financier",
  "Date du versement",
  "BOP",
]
`;
